src/data/wmr/WmrDataSetD2Repository.ts

```diff
@@ -0,0 +1,33 @@
+import { WmrDataSet } from "../../domain/entities/wmr/entities/WmrDataSet";
+import { WmrDataSetRepository } from "../../domain/entities/wmr/repositories/WmrDataSetRepository";
+import { Instance } from "../../domain/instance/entities/Instance";
+import { D2Api } from "../../types/d2-api";
+import { getD2APiFromInstance } from "../../utils/d2-utils";
+import { dataSetFields } from "./WmrSettingsD2Repository";
+
+export class WmrDataSetD2Repository implements WmrDataSetRepository {
+    private api: D2Api;
+    constructor(private instance: Instance) {
+        this.api = getD2APiFromInstance(this.instance);
+    }
+
+    async getById(id: string): Promise<WmrDataSet> {
+        const response = await this.api.models.dataSets
```

Comment:
WmrDataSetD2Repository:: getById and  WmrSettingsD2Repository::getCountrySyncDataElements seem to be doing similar things, would it be a good idea to create a  parameterised util function which can be reused?